src/gt4py/ir/nodes.py

```diff
@@ -685,8 +685,6 @@ class AxisBound(Node):
     level = attribute(of=UnionOf[LevelMarker, VarRef])
     offset = attribute(of=int, default=0)
     loc = attribute(of=Location, optional=True)
-    name: str
-    index: List
```

Comment:
Unrelated fix: I couldn't find where these ever get set. They are referenced when using a `VarRef` for the vertical interval, which isn't even fully supported yet, so I am taking this out. `mypy` complains if these lines are removed, but the usage is not, so that is also removed.

Reply:
Yes, it makes sense since this was never fully implemented. Make sure to update the module docstring with these changes in the `AxisBound` node

Reply:
There was no reference in the module docstring.